src/tools/clippy/tests/ui/postfix_match.rs

```diff
@@ -0,0 +1,77 @@
+#![feature(postfix_match)]
+#![allow(clippy::redundant_at_rest_pattern, clippy::useless_vec)]
+#![warn(
+    clippy::match_on_vec_items, 
+    clippy::single_match, 
+    clippy::match_bool, 
+    clippy::match_as_ref,
+    clippy::needless_match,
+    clippy::match_str_case_mismatch,
+    clippy::redundant_guards
+)]
+
+fn main() {
+    // match_on_vec_items
+    let arr = vec![0, 1, 2, 3];
+    let idx = 1;
+
+    arr[idx].match {
+        //~^ ERROR: indexing into a vector may panic
+        //~| NOTE: `-D clippy::match-on-vec-items` implied by `-D warnings`
+        0 => println!("{}", 0),
+        1 => println!("{}", 3),
+        _ => {},
+    };
+
+    // single_match
+    let x = Some(1u8);
+
+    x.match {
+        Some(y) => {
+            println!("{:?}", y);
+        },
+        _ => (),
+    };
+
+    // match_bool
+    let test: bool = true;
+
+    test.match {
+        //~^ ERROR: you seem to be trying to match on a boolean expression
+        true => 0,
+        false => 42,
+    };
+
+    // match_as_ref
+    let owned: Option<()> = None;
+    let borrowed: Option<&()> = owned.match {
+        None => None,
+        Some(ref v) => Some(v),
+    };
+
+    // needless_match
+    let i = 10;
+    let _: i32 =  i.match {
+        0 => 0,
+        1 => 1,
+        2 => 2,
+        _ => i,
+    };
+
+    // match_str_case_mismatch
+    let var = "BAR";
+    var.to_ascii_lowercase().as_str().match {
+        "foo" => {},
+        "bar" => {},
+        _ => {},
+    }
+
+    // redundant_guards
+    struct C(u32, u32);
+
+    let c = C(1, 2);
+    c.match {
+        C(x, y) if 1 == y => ..,
+        _ => todo!(),
+    };
+}
```

Comment:
Similarly to rustfmt, clippy maintainers would probably want this to be submitted to the clippy repo instead of rustc.

Reply:
The rustfmt folks above seem to be fine with the commit here since it's experimental. But I'm moving the Clippy to a PR to the repo for after this PR.